playbooks/troubleshooting/troubleshooting_guide.adoc

```diff
@@ -0,0 +1,719 @@
+      oc replace -f -
+----
+
+Restart the nfs server and restart the S2I build
+ +
+ +
+
+== Quotas and Limitranges
+
+=== Must make a non-zero request for cpu
+
+After creating a quota for for CPU usage inside of a project you receive the
+following error:
+----
+failed to create build pod: Pod "nodejs-example-2-build" is forbidden: must make a non-zero request for cpu since it is tracked by quota.
+----
+
+==== Solution:
+
+There must be a corresponding CPU amount defined in your project limits
+to go along with the defined quota. Click for more information on
+https://docs.openshift.com/enterprise/3.1/dev_guide/quota.html[quotas]
+and
+https://docs.openshift.com/enterprise/3.1/dev_guide/limits.html[limits].
+ +
+ +
+
+=== I have enough ram for another pod but it won't build
+
+Quotas can prevent build if *build machine* + *new pod* is greater than
+quota. Click for more information on https://docs.openshift.com/enterprise/3.1/dev_guide/quota.html[quotas].
+ +
+ +
+
+== Installation Fails...
+
+=== Job for atomic-openshift-master-api.service failed
+
+----
+failed: [master00.ose.example.com] => {"failed": true}
+msg: Job for atomic-openshift-master-api.service failed because the control process exited with error code. See "systemctl status atomic-openshift-master-api.service" and "journalctl -xe" for details.
+
+
+FATAL: all hosts have already failed -- aborting
+----
+
+==== Solution:
+
+The exact cause of this is unknown at this time. You can try to log into
+each master and checking the status and journald entries mentioned in
+the error. Often though, this proves less than fruitful. A potential
+solution has been to log into the first master, start the service
+manually and restart the ansible installer from the beginning. This has
+been known to allow the installer to continue
+ +
+ +
+
+== WebUI Related
+
+=== Web Console Public URL on a different Port
+
+If the client wishes to move the public URL off of 8443 either because
+of a port conflict or other reason you need to edit the
+master-config.yaml
+
+==== Solution
+
+Edit the *master-config.yaml* on each master and replace the following lines with
+the appropriate values:
+
+----
+  publicURL: https://ose.example.com:8443/console/
+  assetPublicURL: https://ose.example.com:8443/console/
+----
+
+Then restart the *atomic-openshift-master-api* service on each master
+ +
+ +
+
+=== UI Redirecting to the URL of the masters instead of the LB
+
+The main cause for this seems to be the installer failing to honor the 
+**openshift_master_cluster_public_hostnam** option for the OSEv3:vars section.
+This results in the **master-config.yaml** file having the **publicURL** 
+being set to the master's FQDN. ex. publicURL: master00.example.com.
+When the LB selects a master to pass the request to, OSE uses this value
+and substitutes the URL in the browser. These values being set incorrectly
+have implications when deploying your Docker registry.
+ +
+ +
+
+=== Intermittent Login issues
```

Comment:
This section only seems to be referring to htpasswd, so would suggest just updating this section to to specify Intermittent Login Issues with HTPasswd.  This example is only showing multi-master services and does not differentiate from single master service. And I would also suggest removing the part about standard lab instructions.